CMakeLists.txt

```diff
@@ -65,7 +65,7 @@ include(src/CMakeLists.txt)
 include(third-party/CMakeLists.txt)
 include(common/utilities/CMakeLists.txt)
 
-target_link_libraries( ${LRS_TARGET} PRIVATE utilities )
+target_link_libraries( ${LRS_TARGET} PUBLIC utilities )
```

Comment:
Why does the user app must have `utilities`?
It is statically compiled into the SDK no?
I am not sure it's a good idea, where does it effect the user?

Reply:
I can remove this but then you have to add utilities as a dependency in all the other projects -- possibly including users. If they don't use utilities, it won't make a difference.

Reply:
But if we make it a statically dependence of the `librealsense2` library all tools/examples/users will get it.
Only tools using the Utilities explicitly will need it and that's OK.

Reply:
But how does it hurt? Projects that depend on us will either use our CMake in which case it will be transparent, or they compile it separately which will be transparent.

Let's do this and then I'll take some time to try different ways (I wanted to, anyway) to use libreasense and see what's better...

Reply:
I don't understand why you insist for this change..
reverting this change should be easy.

I don't know what can happen for debians / Windows installer and it's not necessary.
We want to minimize the user dependencies.
Integrating a DDS third party took me a month because of this kind of things.

Reply:
Because it simplifies all the rest of our projects.
I'll test it out to make sure.

Reply:
This is a CMake dependency. Nothing else.
This doesn't change ANY of our collaterals...

Reply:
It adds all projects including the user project a linkage dependency.
I prefer that who needs it link to it.
SDK for sure, tools/examples/user if they use it.

Reply:
You understand that this doesn't change anything, right? It doesn't change what's in our .lib or .dll or .so in any way, right?
It's only so that, in CMake, a project that depends on librealsense2 will get a dependency on utilities, which is compiled alongside librealsense2. Only in CMake. It's a convenience.
But I'll remove it and re-add utilities to all our other projects if you prefer...

Reply:
When we install rs-data-collect, 
It will look for utilities library
Why do we want this?
![image](https://user-images.githubusercontent.com/64067618/205491589-2204c011-2252-46a8-b76b-88775c7bb0d0.png)

Reply:
That's not installing it. That's the dependency that CMake adds: not to the executable, but to the build. This is fine. This is a convenience. It lets you include utilities header without having to worry about it.
It doesn't add any dependency to the final executable.
Again: this has effect only in CMake projects. Not on the final output.

Reply:
Because it is static?

Reply:
Right, if it wasn't static it'd be needed at runtime and not just compile-time.

But:
> static libraries do not link other static libraries into the code

Visual Studio has an option:
![image](https://user-images.githubusercontent.com/54848706/205495138-2f305343-315e-41b8-9e6a-b05982e5ecaf.png)

This is not currently on, nor is there any easy way to turn it on via CMake and cross-platform (from limited reading about it).

That means our current code (since introducing utilities/realdds) will have issues if librealsense2 (as a library) is used independently and not via CMake: the user would have to find the utilities lib (or realdds) and link with it, too.

With CMake, making the dependency `PUBLIC` solves this issue: it'll be automatic for the user.

Without CMake, they'll need to manually add the dependency even if they do not use utilities directly: we will likely have to distribute utilities and realdds libraries alongside our realsense2.lib for this use-case. Or we can try to find a solution for Linking library dependencies into realsense2.lib.

I want to add code to our CI that simulates an example that's not part of our main CMake. It would have caught a change like this.

Let's talk about it tomorrow.